### ISSUE TITLE
Since updating to commit 2151ac2181 the overview screen doesn't update

### ISSUE THREAD

**Issue by pjsg** [mcguirepr89/BirdNET-Pi#429]
**Describe the bug**
I updated to commit 2151ac2181 (current main) and the overview screen continues to show a currently analyzing from a couple of days ago:

![image](https://user-images.githubusercontent.com/1508813/177884714-b065af86-0980-42ec-9252-95f41215a7bd.png)

I checked, and the directory for July 05 does not exist:

```
birdnetpi:~/BirdSongs/July-2022$ ls -l
total 300
drwxrwxr-x 2 philip philip 286720 Jul  7 18:51 06-Wednesday
drwxr-xr-x 2 philip philip  12288 Jul  7 18:48 06-Wednesday-Analyzed
drwxr-xr-x 2 philip philip   4096 Jul  7 18:50 07-Thursday
birdnetpi:~/BirdSongs/July-2022$ 
```

The log also shows that processing stops. I tried running `restart_services.sh` and it ran for a few minutes, but then stopped again. I have no idea if this is the same issue as above, but it seems suspicious.

```
Jul 07 18:50:59 python3[45033]: ANALYZING AUDIO... DONE! Time 1.6 SECONDS
Jul 07 18:50:59 python3[45033]: WRITING RESULTS TO /BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-00:32:14.wav.csv ... DONE! WROTE 0 RESULTS.
Jul 07 18:50:59 python3[45033]: READING AUDIO DATA... DONE! READ 5 CHUNKS.
Jul 07 18:50:59 python3[45033]: Date-time: 2022-07-06 00:32:29
Jul 07 18:51:01 birdnet_analysis.sh[47617]: 0.0;3.0-('Branta bernicla_Brant', 0.0736915)
Jul 07 18:51:01 birdnet_analysis.sh[47617]: 3.0;6.0-('Branta canadensis_Canada Goose', 0.09595219)
Jul 07 18:51:01 birdnet_analysis.sh[47617]: 6.0;9.0-('Coccyzus erythropthalmus_Black-billed Cuckoo', 0.064725876)
Jul 07 18:51:01 birdnet_analysis.sh[47617]: 9.0;12.0-('Scolopax minor_American Woodcock', 0.04195895)
Jul 07 18:51:01 birdnet_analysis.sh[47617]: 12.0;15.0-('Aegolius acadicus_Northern Saw-whet Owl', 0.059650008)
Jul 07 18:51:01 birdnet_analysis.sh[45097]: RECORDING_LENGTH set to 15
Jul 07 18:51:01 birdnet_analysis.sh[45097]: /BirdNET-Pi/birdnet/bin/python3 /BirdNET-Pi/scripts\analyze.py --i /BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-00:32:44.wav --o /BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-00:32:44.wav.csv --lat 43.XX --lon -70.XX --week 25 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7 --birdweather_id "IN_USE"
Jul 07 18:51:01 birdnet_recording.sh[45082]: Aborted by signal Terminated...
Jul 07 18:51:01 systemd[1]: Stopping BirdNET Recording...
Jul 07 18:51:01 systemd[1]: birdnet_recording.service: Succeeded.
Jul 07 18:51:01 systemd[1]: Stopped BirdNET Recording.
Jul 07 18:51:01 systemd[1]: birdnet_recording.service: Consumed 2.252s CPU time.
Jul 07 18:51:01 systemd[1]: Stopping BirdNET Analysis...
Jul 07 18:51:01 systemd[1]: birdnet_analysis.service: Succeeded.
Jul 07 18:51:01 systemd[1]: Stopped BirdNET Analysis.
Jul 07 18:51:01 systemd[1]: birdnet_analysis.service: Consumed 19.000s CPU time.
Jul 07 18:51:01 systemd[1]: Stopping BirdNET Analysis Server...
Jul 07 18:51:02 python3[45033]: ANALYZING AUDIO...
Jul 07 18:51:02 systemd[1]: birdnet_server.service: Succeeded.
Jul 07 18:51:02 systemd[1]: Stopped BirdNET Analysis Server.
Jul 07 18:51:02 systemd[1]: birdnet_server.service: Consumed 3min 27.585s CPU time.
```

**System**
I'm running on a 4GB Pi4. 

**To Reproduce**
I have no idea how to reproduce this.

**Comment by mcguirepr89** [comment]
I made a crappy bug `stop_core_services.sh` which stops the `chart_viewer.service` which isn't restarted with `restart_services.sh` :/ sorry about that!!!

You should be able to fix this with "Tools" > "Service Controls" and then Restart "Chart Viewer" service and that ought to do it.

Let me know if that works for you -- if not, I'll be happy to help look into this further with some log troubleshooting

My best,
Patrick

**Comment by pjsg** [comment]
I tried that, but....

```
Jul 07 20:03:28 systemd[1]: Started BirdNET-Pi Chart Viewer Service.
Jul 07 20:03:35 python3[53221]: Traceback (most recent call last):
Jul 07 20:03:35 python3[53221]:   File "/usr/local/bin/daily_plot.py", line 67, in <module>
Jul 07 20:03:35 python3[53221]:     norm = plt.Normalize(confmax.values.min(), confmax.values.max())
Jul 07 20:03:35 python3[53221]:   File "/BirdNET-Pi/birdnet/lib/python3.9/site-packages/numpy/core/_methods.py", line 43, in _amin
Jul 07 20:03:35 python3[53221]:     return umr_minimum(a, axis, None, out, keepdims, initial, where)
Jul 07 20:03:35 python3[53221]: ValueError: zero-size array to reduction operation minimum which has no identity
Jul 07 20:03:36 systemd[1]: chart_viewer.service: Main process exited, code=exited, status=1/FAILURE
Jul 07 20:03:36 systemd[1]: chart_viewer.service: Failed with result 'exit-code'.
Jul 07 20:03:36 systemd[1]: chart_viewer.service: Consumed 8.724s CPU time.
```

**Comment by pjsg** [comment]
Part of the problem appears to be the `cleanup.sh` script which stops the services if it finds more than 10 accumulated files. I have 5083!

Actually, after running `restart_services.sh` a couple of times, it came down to 4950...

While I can probably find all of these files to delete them, this is a pretty nasty trap to fall into....

**Comment by mcguirepr89** [comment]
Hi, @pjsg 

Could you please provide the output from:

```
cat ~/BirdNET-Pi/scripts/cleanup.sh
```

and

```
find $RECS_DIR -path $PROCESSED -prune -o -path $EXTRACTED -prune -o -type f -print
```

**Comment by pjsg** [comment]
```
birdnetpi:~/BirdNET-Pi$ cat ~/BirdNET-Pi/scripts/cleanup.sh
#!/usr/bin/env bash
source /etc/birdnet/birdnet.conf
set -x

cd "${PROCESSED}" || exit 1
empties=($(find ${PROCESSED} -size 57c))
for i in "${empties[@]}";do
  rm -f "${i}"
  rm -f "${i/.csv/}"
done

if [[ "$(find ${PROCESSED} | wc -l)" -ge 100 ]];then
  ls -1t . | tail -n +100 | xargs -r rm -vv
fi

accumulated_files=$(find $RECS_DIR -path $PROCESSED -prune -o -path $EXTRACTED -prune -o -type f -print | wc -l)
[ $accumulated_files -ge 10 ] && stop_core_services.sh
echo "$(date "+%b  %e %I:%M:%S") Stopped Core Services -- It looks like analysis stopped. Check raw recordings in $RECS_DIR and check the birdnet_analysis.service and birdnet_server.service \"journalctl -eu birdnet_analysis -u birdnet_server\"" | sudo tee -a /var/log/syslog
birdnetpi:~/BirdNET-Pi$ 
```

```birdnetpi:~/BirdNET-Pi$ find $RECS_DIR -path $PROCESSED -prune -o -path $EXTRACTED -prune -o -type f -print | head -n 40
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-01:38:30.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-03:47:16.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-15:09:36.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-13:26:51.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-06:34:32.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-01:45:30.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-18:49:38.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-12:51:35.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-05:50:02.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-04:22:01.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-18:01:08.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-10:39:04.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-19:35:39.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-13:43:21.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-18:22:23.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-10:39:19.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-18:44:38.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-05:20:02.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-16:06:22.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-15:49:07.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-18:55:23.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-11:47:35.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-14:15:36.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-20:54:09.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-16:55:22.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-11:03:49.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-16:48:07.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-13:54:21.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-18:17:08.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-17:32:53.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-01:34:15.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-17:27:23.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-19:30:54.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-02:17:45.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-21:06:54.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-02:34:15.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-13:36:06.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-18:36:38.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-12:49:20.wav
/home/philip/BirdSongs/July-2022/06-Wednesday/2022-07-06-birdnet-06:25:02.wav
birdnetpi:~/BirdNET-Pi$ find $RECS_DIR -path $PROCESSED -prune -o -path $EXTRACTED -prune -o -type f -print | wc
   4885    4885  381044
birdnetpi:~/BirdNET-Pi$ 
```

```
birdnetpi:~/BirdNET-Pi$ git show
commit 2151ac2181f2e675d85a103b5dc110c4b677c45c (HEAD -> main, origin/main, origin/HEAD)
Author: ehpersonal38 <103586016+ehpersonal38@users.noreply.github.com>
Date:   Wed Jul 6 16:31:59 2022 -0400

    no need to check the length

diff --git a/homepage/views.php b/homepage/views.php
index 32f53d6..d446f63 100644
--- a/homepage/views.php
+++ b/homepage/views.php
@@ -6,9 +6,7 @@ $home = shell_exec("awk -F: '/1000/{print $6}' /etc/passwd");
 $home = trim($home);
 if(!isset($_SESSION['behind'])) {
   $fetch = shell_exec("sudo -u".$user." git -C ".$home."/BirdNET-Pi fetch 2>&1");
-  if(strlen($fetch) > 0) {
-    $_SESSION['behind'] = trim(shell_exec("sudo -u".$user." git -C ".$home."/BirdNET-Pi status | sed -n '2 p' | cut -d ' ' -f 7"));
-  }
+  $_SESSION['behind'] = trim(shell_exec("sudo -u".$user." git -C ".$home."/BirdNET-Pi status | sed -n '2 p' | cut -d ' ' -f 7"));
   if(isset($_SESSION['behind'])&&intval($_SESSION['behind']) >= 99) {?>
   <style>
   .updatenumber { 
diff --git a/scripts/system_controls.php b/scripts/system_controls.php
index 512784f..335444c 100644
--- a/scripts/system_controls.php
+++ b/scripts/system_controls.php
@@ -5,9 +5,7 @@ $user = trim($user);
 $home = shell_exec("awk -F: '/1000/{print $6}' /etc/passwd");
 $home = trim($home);
 $fetch = shell_exec("sudo -u".$user." git -C ".$home."/BirdNET-Pi fetch 2>&1");
-if(strlen($fetch) > 0) {
-  $_SESSION['behind'] = trim(shell_exec("sudo -u".$user." git -C ".$home."/BirdNET-Pi status | sed -n '2 p' | cut -d ' ' -f 7"));
-}
+$_SESSION['behind'] = trim(shell_exec("sudo -u".$user." git -C ".$home."/BirdNET-Pi status | sed -n '2 p' | cut -d ' ' -f 7"));
 ?><html>
 <meta name="viewport" content="width=device-width, initial-scale=1">
 <br>
birdnetpi:~/BirdNET-Pi$ 
```

**Comment by mcguirepr89** [comment]
Alright -- this should fix your installation:

1. ```
   sudo sed -i 's|\*/3|#*/3|' /etc/crontab
   ```
1. ```
   restart_services.sh
   ```
1. Then watch as the files disappear from this command:
   ```
   source /etc/birdnet/birdnet.conf
   watch find $RECS_DIR -path $PROCESSED -prune -o -path $EXTRACTED -prune -o -type f -print
   ```
1. Once that list is handled, you can press "Ctrl + C" to stop the `watch` command and then
1. Put the crontab back in place:
   ```
   sudo sed -i 's|#\*/3|*/3|' /etc/crontab
   ```


Let me know how that goes for you -- sorry for all of this trouble caused by my careless bug. I shouldn't be a problem after this.

My best,
Patrick

**Comment by ehpersonal38** [comment]
Reposting my comment from closed issue #428:

Having this same exact problem since commit ce017a9baea92169f44a9ed5d277dd777ada9901. The problem is that `scripts/cleanup.sh` is detecting a backlog 10 or more files in `/home/pi/BirdSongs/Processed`, and subsequently stopping all services. But a few lines above that script, it only ever deletes files from Processed if theres 100 or more? Can you shed any light @mcguirepr89 ?

For now I'm commenting out the last 3 lines in `scripts/cleanup.sh` and then running `bash ~/BirdNET-Pi/scripts/restart_services.sh`

**Comment by mcguirepr89** [comment]
Sorry, I've been out of town.

I will just remove the section that stops services when it detects a backlog -- most folks will just have to figure it out themselves -- obviously putting the syslog comment isn't enough to have folks asses why their system has a backlog. Not a great commit, I suppose. I'll remove it now.
